pravega/pravega
# Issue 3115 : fix javadoc warnings

**Change log description**  
Fixes build warnings/errors during javadocJar task by fixing classpath in gradle task and updating javadocs that were malformed.

**Purpose of the change**  
Fixes #3115

**What the code does**  
This is a javadoc only change, 
1. Classpath was missing in generateJavadoc task, adding it fixes most of the issues.
2. A few classes had old method signatures in `{@link }` directive, fixing these.
3. Fixes formatting issues related to code snippets containing generics and lambdas

**How to verify it**  
`./gradelw javadocJar`  should produce significantly reduced number of warnings
This is a javadoc only change there should be no functional changes. 
Also note that this change does not fix ALL warnings or errors. But it does fix majority of them. There are still a few (<25) warnings that still won't be fixed by this code as they might require some more thorough solution to work properly.
